src/tests/regression/reported/issue-552-send_by_ref-singleton.f90

```diff
@@ -5,29 +5,35 @@ program main
       real, allocatable :: a(:, :, :)[:], b(:, :, :)
    end type
 
-   integer :: me, nimg
+   integer :: me, remote, nimg
    type(co_obj) :: co
 
    me = this_image()
    nimg = num_images()
+   remote = merge(2, 1, me == 1)
 
    if (nimg /= 2) error stop 1
 
-   allocate(co % a(1, 10, 20)[*], co % b(1, 10, 20))
+   allocate(co % a(1, 4, 8)[*], source=99.)
+   allocate(co % b(1, 4, 8))
    call random_number(co % b)
+   sync all
 
-   ! singleton on 1st dimension (delta == 1)
-   co % a(1, :, :)[merge(2, 1, me == 1)] = co % b(1, :, :)
+   ! singleton on the 1st dimension (delta == 1), triggers the bug
+   co % a(1, :, :)[remote] = co % b(1, :, :)
 
-   ! icar output
-   ! 3/4: Entering send_by_ref(may_require_tmp = 1, dst_type = 3).
-   ! 3/4: _gfortran_caf_send_by_ref() remote_image = 3, offset = 0, remote_mem = 0x7f4a29422060
-   ! 3/4: _gfortran_caf_send_by_ref() remote desc rank: 3 (ref_rank: 7297736)
-   ! 3/4: _gfortran_caf_send_by_ref() remote desc dim[0] = (lb = 1, ub = 1, stride = 1)
-   ! 3/4: _gfortran_caf_send_by_ref() remote desc dim[1] = (lb = 1, ub = 20, stride = 1)
-   ! 3/4: _gfortran_caf_send_by_ref() remote desc dim[2] = (lb = 1, ub = 103, stride = 20)
-   ! 3/4: _gfortran_caf_send_by_ref() extent(dst, 0): 1 != delta: 20.  ! <====== mistmatch src_cur_dim not incremented
-   ! 3/4: _gfortran_caf_send_by_ref() extent(dst, 1): 20 != delta: 101.
+   sync all
+   if (any(abs(co % b(1, :, :) - co % a(1, :, :)[remote]) > epsilon(0.))) error stop 1
+   if (me == 1) write(*, *) 'Test passed.'
 
-   write(*, *) 'Test passed.'
 end program
+
```

Comment:
Changes to this test [seem to have broken it](https://travis-ci.org/sourceryinstitute/OpenCoarrays/jobs/395684332#L3076) when running a debug build test with GFortran 7.3 on Ubuntu.  (The [error from the previous commit](https://travis-ci.org/sourceryinstitute/OpenCoarrays/jobs/395643539#L3334) is a different, unrelated issue.) Does it require the latest changes in GFortran 8 and >= 7.4?